osu.Game/Storyboards/IStoryboardElementWithDuration.cs

```diff
@@ -12,9 +12,17 @@ public interface IStoryboardElementWithDuration : IStoryboardElement
     {
         /// <summary>
         /// The time at which the <see cref="IStoryboardElement"/> ends.
+        /// This is consumed to extend the length of a storyboard to ensure all visuals are played to completion.
         /// </summary>
         double EndTime { get; }
 
+        /// <summary>
+        /// The time this element displays until.
+        /// This is used for lifetime purposes, and includes long playing animations which don't necessarily extend
+        /// a storyboard's play time.
+        /// </summary>
+        double EndTimeForDisplay { get; }
```

Comment:
Not sure on the naming here... Thoughts on doing something like

- `EndTime` => `EarliestEndTime` (to reflect that it's the "earliest" time that the element is allowed to terminate being animated at)
- `EndTimeForDisplay` => `CompleteEndTime`?

Probably would need to adjust `StoryboardElementExtensions.GetEndTime()` to be called `GetEarliestEndTime()` instead or something.